source/runtime.cpp

```diff
@@ -4794,31 +4795,64 @@ static std::string expand_macro_string(const std::string &input, std::vector<std
 	return result;
 }
 
-void reshade::runtime::save_screenshot(const std::string_view postfix)
-{
-	const unsigned int screenshot_count = _screenshot_count;
-
-	std::string screenshot_name = expand_macro_string(_screenshot_name, {
-		{ "AppName", g_target_executable_path.stem().u8string() },
+void reshade::runtime::save_screenshot(const std::string_view postfix) {
+  const unsigned int screenshot_count = _screenshot_count;
+  std::filesystem::path _current_screenshot_path;
+    if (_screenshot_path_split_appname) {
+    std::string _screenshot_name_split =
+        (_screenshot_name == "%AppName% %Date% %Time%_%TimeMS%" ||
+         _screenshot_name == "%AppName% %Date% %Time%")
+            ? "%Date% %Time%_%TimeMS%"
+            : _screenshot_name;
+	 // This will remove the appname and leading whitespace if people have not changed their target path but select this option. Also updates to new default
+        // if (_screenshot_name == "%AppName% %Date% %Time%_%TimeMS%" || _screenshot_name == "%AppName% %Date% %Time)  _screenshot_name = "%Date% %Time%_%TimeMS%";
```

Comment:
Great idea! But if you want more powerful could we make one little amendment to the macro tooltip to add %BeforeAfter% as a macro? I have updated the tooltip hash number for all languages so they still function but havent added any localizations. IMO its too useful not to include and I think it fits in well to the existing menu

![image](https://github.com/user-attachments/assets/23985893-e6cb-474c-9099-600f858f18d2)

As for the main implementation, I'm not sure if this is the most optimal handling as string manipulation in C++ is not my favorite, but at least from a functionality standpoint its working great and I think matches your intent. I added some handling to eliminate trailing and leading whitespaces and it generally does what its supposed to. The only outstanding issue is that Having `\\%BeforeAfter%\\`  in your path and then disabling the option to save before and after images leads to a double slash despite a brief attempt to handle it. However you can type an indefinite number of slashes in the directory boxes and they're ignored, so ultimately not a huge problem.  Everything aside from this code block is just some minor changes

`
void reshade::runtime::save_screenshot(const std::string_view postfix) {
  const unsigned int screenshot_count = _screenshot_count;
//instantiate reusable vector excluding count for path because that would be literally insane to want that
 std::vector<std::pair<std::string, std::string>> macros = {
      {"BeforeAfter", std::string(postfix)  },
      {"AppName", g_target_executable_path.stem().u8string()},
#if RESHADE_FX
      {"PresetName", _current_preset_path.stem().u8string()},
#endif
    };

  // Expand macros in path
  std::filesystem::path expanded_path  = std::filesystem::u8path(  expand_macro_string(_screenshot_path.u8string(), macros));


#if RESHADE_FX
 macros.push_back({"Count", std::to_string(screenshot_count)});
#endif

  // Expand macros in filename with count included
  std::string screenshot_name = expand_macro_string(_screenshot_name, macros);


  // trim leading and trailing whitespaces from the name
  screenshot_name.erase(0, screenshot_name.find_first_not_of(" \t\n\r\f\v"));
  screenshot_name.erase(screenshot_name.find_last_not_of(" \t\n\r\f\v") + 1);


screenshot_name += (_screenshot_format == 0 ? ".bmp" : _screenshot_format == 1 ? ".png" : ".jpg");

const std::filesystem::path screenshot_path = g_reshade_base_path / expanded_path  / std::filesystem::u8path(screenshot_name);

`

I'll push these commits along in just a moment but I have a few more things ready or almost ready that I'll try to catch you in discord to discuss